draft-ietf-tls-tls13.md

```diff
@@ -1854,14 +1820,6 @@ subsequent connections.
 The "key_share" extension contains the endpoint's cryptographic parameters
 for non-PSK key establishment methods (currently DHE or ECDHE).
 
-Clients which offer one or more (EC)DHE cipher suites MUST send this
-extension and SHOULD send at least one supported KeyShareEntry value.
-Servers MUST NOT negotiate any of these cipher suites unless a supported
-value was provided.
-If this extension is not provided in a ServerHello or ClientHello,
```

Comment:
Right, hence just the ServerHello clause. ClientHello clause was always kind of silly. But when receiving a ServerHello, if the server PSK extension says `psk_dhe_ke` or is missing, then we should fail the connection if `key_share` is missing. I.e. there is redundancy in the ServerHello.